File: 프로그래머스/1/42748. K번째수/K번째수.java
import java.util.*;

class Solution {
    public int[] solution(int[] array, int[][] commands) {
        List<Integer> answerList = new ArrayList<>();
        
        for(int[] command : commands){
            List<Integer> list = new ArrayList<>();
            for(int i=command[0]-1; i<command[1]; i++){
                list.add(array[i]);
            }
            Collections.sort(list);
            answerList.add(list.get(command[2]-1));
        }
        
        // List<Integer> -> int[] 변환 (stream 사용)
        return answerList.stream().mapToInt(Integer::intValue).toArray();
    }
}

File: 프로그래머스/1/42748. K번째수/K번째수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> array, vector<vector<int>> commands) {
    vector<int> answer;
    
    vector<int> tmp;
    for(vector<int> command : commands){
        tmp.clear();
        for(int j=command[0]-1; j<=command[1]-1; j++) tmp.push_back(array[j]);
        sort(tmp.begin(), tmp.end());
        answer.push_back(tmp[command[2]-1]);
    }
    
    return answer;
}